gobblin-compaction/src/main/java/gobblin/compaction/action/CompactionCompleteFileOperationAction.java

```diff
@@ -43,36 +47,60 @@ public CompactionCompleteFileOperationAction (State state, CompactionAvroJobConf
    * Replace the destination folder with new output from map-reduce job
    * and create a file for next run record count comparison.
    */
-  public void onCompactionJobComplete (FileSystemDataset dataset) {
+  public void onCompactionJobComplete (FileSystemDataset dataset) throws IOException {
     if (configurator != null && configurator.isJobCreated()) {
       CompactionPathParser.CompactionParserResult result = new CompactionPathParser(state).parse(dataset);
       Path tmpPath = configurator.getMrOutputPath();
       Path dstPath = new Path (result.getDstAbsoluteDir());
 
-      try {
-        // get record count from map reduce job counter
-        Job job = this.configurator.getConfiguredJob();
-        Counter counter = job.getCounters().findCounter(AvroKeyMapper.EVENT_COUNTER.RECORD_COUNT);
-        long recordCount = counter.getValue();
+      // get record count from map reduce job counter
+      Job job = this.configurator.getConfiguredJob();
+      Counter counter = job.getCounters().findCounter(AvroKeyMapper.EVENT_COUNTER.RECORD_COUNT);
+      long recordCount = counter.getValue();
 
-        // move output from mapreduce to final destination defined by dataset
+
+      boolean renamingRequired = this.state.getPropAsBoolean(MRCompactor.COMPACTION_RENAME_SOURCE_DIR_ENABLED,
```

Comment:
The name is a little confusing here. Maybe add a comment that this is append delta mode due to the compaction rename source dir mode being enabled or rename the variable to appendDeltaOutput.